src/cpp/src/llm_pipeline.cpp

```diff
@@ -339,11 +400,354 @@ std::pair<std::string, Any> generation_config(const GenerationConfig& config) {
+        OPENVINO_THROW("Currently NPU device may only process up to " + std::to_string(m_kvcache_desc.total_size) + " tokens");
+    }
+
+    // NB: Reset tensors on every generate call - chat conversation isn't supported yet!
+    prepareForNewConversation();
+
+    auto padded_input_ids = m_prefill_request.get_tensor("input_ids");
+    copy_with_left_offset(input_ids, padded_input_ids);
+
+    auto padded_attention_mask = m_prefill_request.get_tensor("attention_mask");
+    copy_with_left_offset(attention_mask, padded_attention_mask);
+
+    auto padded_position_ids = m_prefill_request.get_tensor("position_ids");
+    auto* padded_pos_data = padded_position_ids.data<int64_t>();
+    std::iota(padded_pos_data + (m_kvcache_desc.total_size - prompt_len + 1), padded_pos_data + padded_position_ids.get_size(), 0u);
+
+    m_prefill_request.infer();
+
+    // NB: Now there are prompt_len tokens in KV-cache
+    m_kvcache_desc.num_stored_tokens += prompt_len;
+    int64_t last_token = utils::argmax(m_prefill_request.get_tensor("logits"), 0);
+    if (streamer_ptr && streamer_ptr->put(last_token)) {
+        return results;
+    }
+
+    padded_attention_mask.copy_to(m_kvcache_request.get_tensor("attention_mask"));
+    // NB: Setup KV-cache model for execution
+    const auto& kvcache_compiled = m_kvcache_request.get_compiled_model();
+    for (auto input : kvcache_compiled.inputs()) {
+        // NB: Remapping input/output names
+        const std::string input_pattern = "past_key_values";
+        const std::string output_pattern = "present";
+        const auto input_name = input.get_any_name();
+        auto output_name = input_name;
+        const auto pos = output_name.find(input_pattern);
+        if (pos == std::string::npos) {
+            continue;
+        }
+        output_name.replace(pos, input_pattern.length(), output_pattern);
+        // NB: Points input and output KV-cache tensors to the same buffer
+        m_kvcache_request.set_tensor(input_name, m_kvcache_request.get_tensor(output_name));
+        // NB: Copy KV-cache from prefill model to KV-cache model
+        m_prefill_request.get_tensor(output_name).copy_to(m_kvcache_request.get_tensor(input_name));
+    }
+
+    auto* input_ids_data = m_kvcache_request.get_tensor("input_ids").data<int64_t>();
+    auto* position_ids_data = m_kvcache_request.get_tensor("position_ids").data<int64_t>();
+    auto* attention_mask_data = m_kvcache_request.get_tensor("attention_mask").data<int64_t>();
+
+    const size_t max_tokens = config.get_max_new_tokens(prompt_len);
+    for (int i = 0; i < max_tokens - 1; ++i) {
+        input_ids_data[0] = last_token;
+        position_ids_data[0] = m_kvcache_desc.num_stored_tokens;
+        attention_mask_data[m_kvcache_desc.total_size - m_kvcache_desc.num_stored_tokens - 1] = 1u;
+
+        m_kvcache_request.infer();
+        m_kvcache_desc.num_stored_tokens += 1;
+
+        last_token = utils::argmax(m_kvcache_request.get_tensor("logits"), 0);
```

Comment:
what's about other sampling parameters?

Reply:
well, I thing that's it for `greedy` search which is the only one that currently supported for NPU pipeline.

I think the rest of decoding search algos can be also supported but require a little refactoring to re-use it in NPU pipeline. Didn't touch to minimize changes...